### DISCUSSION TITLE
Fresh Netbox installation 3.5.2 without internet connection

### DISCUSSION THREAD

**Discussion by lcascales** [netbox-community/netbox#12761] [Q&A] [answered]
Hello everyone,

The reason I am trying to install Netbox on a VM without internet connection is because we are trying to have it running and perform tests as the main goal is to present it as a solution to a client's private network , and all configurations and progress should be documented for future reference.

The first obstacle is the execution of the upgrade.sh script, where it will fail because it can't download any packages, so I did a bit of research to figure out a way to obtain the packages to upgrade (pip and the remaining packages) on another VM with internet connection and transfer it to the target machine, in order to install it offline.

However some packages are tar.gz packages and had to install some of those manually before picking up, but later ran into conflicting packages, I don't have access to the work laptop at this point, but will update this thread when I do, to show the output of the pip install command, still I think I missed some lines from the requirements.txt file as I copied the content and pasted on the vi editor on the other VM. Still need to wait for tomorrow to see the logs again.

Regardless of my update, would it be a better alternative to use docker image instead?

Thanks in advance, will update this thread when possible.

**Comment by jcollie** [comment, marked as answer]
Yeah, I think that it would be easier to use the Docker image as it's one "thing" that should have everything you need built right in. It'll definitely be easier maintaining it over the long run too rather than trying to maintain an offline cache of all the individual Python packages.

**Reply by lcascales** [reply]
Yup, that's a pretty solid point
